### PR TITLE
Add pre-commit tool

### PR DESCRIPTION
**Pull request by KAUTH**
In this commit, we add the pre-commit tool to
improve the Software Development Lifecycle
of the project by automatically running checks
(linting & formatting) before one commits their
changes.

Specifically:
- we added a local hook with pre-commit that
calls tox to lint and format
Additionally, the check-added-large-files
commit hook was added (prevents giant
files from being committed).

- We fixed doc build and pylint failing:
* Docs were failing with the pinned versions of
their dependencies (sphinx,
sphinx-autodoc-typehints).
So we unpinned the dependencies (also good
as best practice).
* Pylint was not able to read the sections
related to its configuration in tox.ini.
So we added a pylint section in tox.ini with the
necessary key/value entries.

- We updated the CONTRIBUTING.md with
how to install pre-commit

### CODE REVIEW

**Review by matthieucan — COMMENTED** [review]
Thank you @KAUTH! This looks very good to me :rocket:
Only minor comments below

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -79,4 +79,13 @@ And then run everything with:
 tox
 ```
 
+To automatically run checks before you commit your changes you should install and run **pre-commit**:
+
+```shell
+pip install -U pre-commit
```
[the remark this answers is not in the document]
Comment (KAUTH) [reply CONTRIBUTING.md:85]: Very good point.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        files: 'src/diepvries/'        
+-   repo: local
+    hooks:
+      - id: tox-lint
+        name: run linting & formatting from tox
+        description: Runs linting and formatting from tox
+        entry: tox -e lint
+        language: system
+        files: ''
+        verbose: true
```
[the remark this answers is not in the document]
Comment (KAUTH) [reply .pre-commit-config.yaml:15]: Done

**Review by matthieucan — APPROVED** [review]
Great work! Could you add a line describing the changes in `CHANGELOG.md`?

**Comment by KAUTH** [comment]
Sure thing 😊

**Review by michael-the1 — APPROVED** [review]
🚀

**Comment by matthieucan** [comment]
@KAUTH Thank you for the PR! :rocket:

**Pull request merged**